core/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java

```diff
@@ -87,18 +87,32 @@ protected void masterOperation(final IndicesAliasesRequest request, final Cluste
         List<AliasActions> actions = request.aliasActions();
         List<AliasAction> finalActions = new ArrayList<>();
         boolean hasOnlyDeletesButNoneCanBeDone = true;
+
+        // Resolve all the AliasActions into AliasAction instances and gather all the aliases
         Set<String> aliases = new HashSet<>();
         for (AliasActions action : actions) {
-            //expand indices
             String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(state, request.indicesOptions(), action.indices());
-            //collect the aliases
             Collections.addAll(aliases, action.aliases());
             for (String index : concreteIndices) {
+                if (action.actionType() == AliasActions.Type.REMOVE_INDEX) {
+                    finalActions.add(new AliasAction.RemoveIndex(index));
```

Comment:
> There aren't any aliases to resolve if you are removing the index though. Right now it'd just NPE. I could probably hack it so it'd work but I don't think it'd be too pretty.

Right, I see.

> I think I should just move the switch statement up a level and move the for loop into the two arms that need it? Something like that.

That could work too, but then we would end up with multiple for-loops? I don't think that would be pretty too, so maybe just leave it this way.